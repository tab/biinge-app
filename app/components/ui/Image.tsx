import React from "react"
import { View, Text, StyleProp, StyleSheet } from "react-native"
import Animated, { FadeIn } from "react-native-reanimated"
import FastImage from "react-native-fast-image"

import { imageUrl } from "helpers/imageUrl"
import colors from "styles/colors"
import { TMDBPosterSizeType } from "types"

type Props = {
  title?: string
  path: string
  size: TMDBPosterSizeType
  style?: StyleProp<any>
}

const ImageComponent = ({ title, path, size, style }: Props) => {
  return (
    <Animated.View entering={FadeIn} style={[style, styles.root]}>
      <View style={styles.content}>
        <Text style={styles.text}>{title}</Text>
      </View>
      <FastImage
        style={style}
        source={{
          uri: imageUrl(size, path),
          priority: FastImage.priority.normal,
        }}
        resizeMode={FastImage.resizeMode.cover}
      />
    </Animated.View>
  )
}

export default ImageComponent

const styles = StyleSheet.create({
  root: {
    backgroundColor: colors.americanSilver,
  },
  content: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: 16,
    fontWeight: "600",
    color: colors.white,
    textAlign: "center",
  },
})
