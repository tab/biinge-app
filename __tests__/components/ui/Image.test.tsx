import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Image from "components/ui/Image"

describe("Image", () => {
  const imgPath = "eO3r38fwnhb58M1YgcjQBd3VNcp.jpg"

  it("renders correctly", () => {
    const tree = render(<Image size="w780" path={imgPath} />)
    expect(tree).toMatchSnapshot()
  })
})
