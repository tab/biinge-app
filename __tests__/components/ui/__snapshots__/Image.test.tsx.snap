// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  entering={[Function]}
  style={
    {
      "backgroundColor": "#d1cecf",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "priority": "normal",
          "uri": "https://image.tmdb.org/t/p/w780eO3r38fwnhb58M1YgcjQBd3VNcp.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
